contracts/common/DataAvailability.sol

```diff
@@ -0,0 +1,28 @@
+// (c) Cartesi and individual authors (see AUTHORS)
+// SPDX-License-Identifier: Apache-2.0 (see LICENSE)
+
+pragma solidity ^0.8.8;
+
+import {IInputBox} from "../inputs/IInputBox.sol";
+
+/// @title Data Availability
+/// @notice Defines the signatures of data availability solutions.
+interface DataAvailability {
+    /// @notice The application receives inputs only from
+    /// a contract that implements the `IInputBox` interface.
+    /// @param inputBox The input box contract address
+    function InputBox(IInputBox inputBox) external;
+
+    /// @notice The application receives inputs from
+    /// a contract that implements the `IInputBox` interface,
+    /// and from Espresso, starting from a given block height,
+    /// and for a given namespace ID.
+    /// @param inputBox The input box contract address
+    /// @param fromBlock Height of first Espresso block to consider
+    /// @param namespaceId The Espresso namespace ID
+    function InputBoxAndEspresso(
```

Comment:
Not sure `InputBoxAndEspresso` should be part of a global official `DataAvailability` interface. I am expecting people to be able to add their own without messing with our contracts.

Reply:
I think it's good to have at least the ones we fully support defined here so that deployment front-ends and the node can utilize the same definitions. :-)

Reply:
People can still provide their own DA config blobs on deployment, and hack their own DA provider node plug-in that understands and decodes that blob.

Reply:
Also, if we later need to add a new DA signature, they can just open a PR. It won't affect the bytecode of our deployed contracts, and therefore won't trigger any redeployments. Therefore, we can release minor versions with these changes.

Reply:
Hmm I see. But "fully supporting" an integration is a complicated term, and I'm still not very comfortable with our regular contracts mentioning specific integrations/extensions, I feel it should be separate. Couldn't these signatures be placed in separate interfaces?
I'd like to hear other opinions, maybe @pedroargento or @tuler?

Reply:
We could totally place these definitions in a third place, but it is convenient to place them here because the `rollups-node` repo already uses the artifacts from `rollups-contracts` (which includes contracts/interface ABIs), and front-ends often depend on the `@cartesi/rollups` npm package to interact with our contracts (e.g. when deploying).

Reply:
But yeah, we could create a new repo (e.g. `rollups-da-config`) with just this file, which would not be imported by `rollups-contracts`, but would be imported/used by the node and deployment front-ends.

Reply:
> I'm still not very comfortable with our regular contracts mentioning specific integrations/extensions

Our regular contracts are not mentioning integrations/extensions. If you were to draw a dependency graph of the `rollups-contracts` with this PR, this interface would be isolated from every other contract. It would not be referenced by any other contract, and therefore not influence any of them. That's why we could, in theory, move it to a separate repo.

Reply:
Just to be clear that I'm not advocating to moving it to somewhere else. I think it's okay to keep it here. If this interface grows out of control, we can think about create a dedicated repo for it.